addons/sale/models/sale.py

```diff
@@ -643,7 +643,10 @@ def _create_invoices(self, grouped=False, final=False):
             if not invoiceable_lines:
                 raise UserError(_('There is no invoiceable line. If a product has a Delivered quantities invoicing policy, please make sure that a quantity has been delivered.'))
 
-            invoice_vals['invoice_line_ids'] = [
+            # there is a chance the invoice_vals['invoice_line_ids'] already contains data when
+            # another module extends the method `_prepare_invoice()`. Therefore, instead of
+            # replacing the invoice_vals['invoice_line_ids'], we append invoiceable lines into it
+            invoice_vals['invoice_line_ids'] += [
```

Comment:
You can't do this, as this will fail if the element doesn't exist in the dictionary. Anyway, the way to do this is to append lines AFTER this method, not on `_prepare_invoice`

Reply:
As the `_prepare_invoice()` always gives the `invoice_line_ids` as `[]`. It's safe to apply the operator `+=`
However, do you think I should test if it exists and is a list first?

Reply:
FYI, In my situation, a single sales order line may generate more than one invoice lines. With the current code structure, I have no way to add more invoice lines than except overriding the `_prepare_invoice()`